src/services/refactors/moveToNewFile.ts

```diff
@@ -0,0 +1,359 @@
+            changes.deleteNode(sourceFile, importDecl);
+        }
+        else {
+            if (name && defaultUnused) {
+                changes.deleteNode(sourceFile, name);
+            }
+            if (namedBindings) {
+                if (namedBindingsUnused) {
+                    changes.deleteNode(sourceFile, namedBindings);
+                }
+                else if (namedBindings.kind === SyntaxKind.NamedImports) {
+                    for (const element of namedBindings.elements) {
+                        if (isUnused(element.name)) changes.deleteNodeInList(sourceFile, element);
+                    }
+                }
+            }
+        }
+    }
+
+    function getNewFileImportsAndAddExportInOldFile(
+        oldFile: SourceFile,
+        importsToCopy: ReadonlySymbolSet,
+        newFileImportsFromOldFile: ReadonlySymbolSet,
+        changes: textChanges.ChangeTracker,
+        checker: TypeChecker,
+    ): ReadonlyArray<ImportDeclaration> {
+        const copiedOldImports = mapDefined(oldFile.statements, oldStatement =>
+            isImportDeclaration(oldStatement)
+                ? filterImport(oldStatement, oldStatement.moduleSpecifier, name => importsToCopy.has(checker.getSymbolAtLocation(name)))
+                : undefined);
+
+        // Also, import things used from the old file, and insert 'export' modifiers as necessary in the old file.
+        let oldFileDefault: Identifier | undefined;
+        const oldFileNamedImports: ImportSpecifier[] = [];
+        const markSeenTop = nodeSeenTracker(); // Needed because multiple declarations may appear in `const x = 0, y = 1;`.
+        newFileImportsFromOldFile.forEach(symbol => {
+            for (const decl of symbol.declarations) {
+                if (!isTopLevelDeclaration(decl) || !isIdentifier(decl.name)) continue;
+
+                const top = getTopLevelDeclarationStatement(decl);
+                if (markSeenTop(top) && !hasModifier(top, ModifierFlags.Export)) {
+                    changes.insertExportModifier(oldFile, top);
+                }
+                if (hasModifier(decl, ModifierFlags.Default)) {
+                    oldFileDefault = decl.name;
+                }
+                else {
+                    oldFileNamedImports.push(createImportSpecifier(undefined, decl.name));
+                }
+            }
+        });
+
+        const oldFileImport = makeImportIfNecessary(oldFileDefault, oldFileNamedImports, `./${removeFileExtension(getBaseFileName(oldFile.fileName))}`);
+        return [...copiedOldImports, ...(oldFileImport ? [oldFileImport] : emptyArray)];
+    }
+
+    function makeUniqueModuleName(moduleName: string, extension: string, inDirectory: string, host: LanguageServiceHost): string {
+        while (true) {
+            const name = combinePaths(inDirectory, moduleName + extension);
+            if (!host.fileExists(name)) return moduleName;
+            moduleName += "0";

```

Comment:
nit, i would make that 1-based (more natural) and i would use a separator of sorts. VSCode adds a `.1`, Windows will add `(2)` to duplicate files. 
we could go with the `.1` or we can use `_1`.